plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/PyStringCodeCompletion.java

```diff
@@ -240,6 +245,88 @@ public TokensOrProposalsList getCodeCompletionProposals(CompletionRequest reques
         return ret;
     }
 
+    private TokensOrProposalsList getCompletionsForTypedDict(CompletionRequest request) throws CoreException,
+            BadLocationException, IOException, MisconfigurationException, PythonNatureWithoutProjectException {
+        if (request.isInKeytip) {
```

Comment:
Is the `isInKeytip` still needed? 

i.e.: doesn't the algorithm below already handle all we need (in which case the changes related to computing the `isInKeytip` would not be needed anymore...)?

Reply:
I think it's really not needed at this point, so I deleted it in the last commit ;)